### DISCUSSION TITLE
What is the defference of options "enableMultiSort" and "isMultiSortEvent"

### DISCUSSION THREAD

**Discussion by coder-xiaotian** [TanStack/table#4890] [Q&A] [answered]
I want to use the multi-sort feature, but when I set the option `enableMultiSort:true`, it doesn't work. But when I use `isMultiSortEvent: () => true`, it works well.

It seems the `enableMultiSort:true` is not useful. Is it a bug or a verse option?

**Comment by KevinVandy** [comment]
manualSorting means that all client side sorting is turned off. This is useful if you are already sorting data on your server.

**Reply by coder-xiaotian** [reply]
Thank's for your reply. But I mistook the option, actually is "enableMultiSort", not "manualSorting", I want to ask what is the difference the "enableMultiSort: true" and "isMultiSortEvent: () => true".

**Reply by KevinVandy** [reply, marked as answer]
There's some api docs about this, though guess they could be more clear

https://tanstack.com/table/v8/docs/api/features/sorting#ismultisortevent
https://tanstack.com/table/v8/docs/api/features/sorting#enablemultisort

`enableMultiSort` allows or turns it off completely.
`isMultiSortEvent` allows you to customize when an action causes multi-sort, or when an action just causes to switch to a new single sort. By default, multi-sort is only occurs when the user holds down the `Shift` key while clicking a new column to sort. But if you specify `isMultiSortEvent: (e) => true`, that will override and make any new sort click always a multi-sort event, no extra keys need to be pressed by the user.

**Reply by coder-xiaotian** [reply]
> There's some api docs about this, though guess they could be more clear
> 
> https://tanstack.com/table/v8/docs/api/features/sorting#ismultisortevent https://tanstack.com/table/v8/docs/api/features/sorting#enablemultisort
> 
> `enableMultiSort` allows or turns it off completely. `isMultiSortEvent` allows you to customize when an action causes multi-sort, or when an action just causes to switch to a new single sort. By default, multi-sort is only occurs when the user holds down the `Shift` key while clicking a new column to sort. But if you specify `isMultiSortEvent: (e) => true`, that will override and make any new sort click always a multi-sort event, no extra keys need to be pressed by the user.

Thank you. It's very helpful for me. "enableMultiSort" needs to be used together with the Shift key.

**Reply by coder-xiaotian** [reply]
I have made a [PR](https://github.com/TanStack/table/pull/4891).

**Reply by coder-xiaotian** [reply]
Hi bro! Can you merge this [PR](https://github.com/TanStack/table/pull/4891)?
